pyaedt/modeler/Primitives2D.py

```diff
@@ -148,23 +148,23 @@ def create_rectangle(self, position, dimension_list, is_covered=True, name=None,
         ...                                                     name="MyCircle", matname="Copper")
 
         """
-        szAxis = self.plane2d
-        XStart, YStart, ZStart = self._pos_with_arg(position)
+        axis = self.plane2d
+        x_start, y_start, z_start = self._pos_with_arg(position)
         if self.plane2d == "Z":
-            Height = self._arg_with_dim(dimension_list[1])
-            Width = self._arg_with_dim(dimension_list[0])
+            height = self._arg_with_dim(dimension_list[0])
+            width = self._arg_with_dim(dimension_list[1])
         else:
-            Height = self._arg_with_dim(dimension_list[0])
-            Width = self._arg_with_dim(dimension_list[1])
+            height = self._arg_with_dim(dimension_list[0])
+            width = self._arg_with_dim(dimension_list[1])
```

Comment:
@beliaev-maksim The code in this if/else condition seems the same. Is the if/else condition really necessary or shall we go with:
```
        if self.plane2d == "Z":
            height = self._arg_with_dim(dimension_list[1])
            width = self._arg_with_dim(dimension_list[0])
        else:
            height = self._arg_with_dim(dimension_list[0])
            width = self._arg_with_dim(dimension_list[1])
```

Reply:
good catch. For  XY planes my fix is definitely correct. I need to check RZ, may be in RZ it should be swapped